python/Ganga/GPIDev/Base/Objects.py

```diff
@@ -125,13 +125,17 @@ def __deepcopy__(self, memo=None):
         setattr(obj, '_registry', self._registry)
         return obj
 
-    @synchronised
     def _getParent(self):
         return self._parent
 
-    @synchronised
+    @synchronised  # This will lock the _current_ (soon to be _old_) root object
     def _setParent(self, parent):
-        setattr(self, '_parent', parent)
+        if parent is None:
+            setattr(self, '_parent', parent)
+        else:
+            with parent.lock:  # This will lock the _new_ root object
```

Comment:
Well, `lock` is the external interface what API users can use. `_lock`, given its leading underscore, in an implementation detail.